### DISCUSSION TITLE
Insights Module

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3003] [Feature Requests]
# Insights Module

This would allow visualizing data through customizable dashboards.

```
directus_insights
- id (or use name)
- name (string)
- icon
- refresh (int, supports -1)
- panes (json)
```

## Panes
```
- name
- x
- y
- w
- h
```

### Metric
Shows an aggregate function (eg: sum, avg, min, max, count) value based on a subset of data.

- Color
- Collection
- Field
- Function
- Unit Prefix
- Unit Suffix
- Sort Field
- Sort Direction
- Filters
- Limit

### List
Shows a table of items based on filters.

### Timeline
Shows field data values (y axis) mapped against a date/time field (x axis)

![Directus Insights — Empty](https://user-images.githubusercontent.com/522079/93916028-e4cc0c00-fcd6-11ea-884b-3dab5d8f57fd.png)

**Comment by botzill** [comment]
Is this module work in progress?

**Reply by benhaynes** [reply]
Only in the Discovery/Design/Strategy stage, it's a PoC that will be "in work" once it becomes a GitHub issue.

**Comment by aviNrich** [comment]
Hi,
Any progress with that?
Thanks

**Comment by rijkvanzanten** [comment]
This exists now!
